homeassistant/components/syncthru/__init__.py

```diff
@@ -1,23 +1,71 @@
 """The syncthru component."""
 
+from typing import Set, Tuple
+
+from pysyncthru import SyncThru
+
 from homeassistant.components.sensor import DOMAIN as SENSOR_DOMAIN
 from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_URL
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers import aiohttp_client, device_registry as dr
 from homeassistant.helpers.typing import ConfigType, HomeAssistantType
 
+from .const import DEFAULT_MODEL, DEFAULT_NAME_TEMPLATE, DOMAIN
+from .exceptions import SyncThruNotSupported
+
 
 async def async_setup(hass: HomeAssistantType, config: ConfigType) -> bool:
     """Set up."""
+    hass.data.setdefault(DOMAIN, {})
     return True
 
 
 async def async_setup_entry(hass: HomeAssistantType, entry: ConfigEntry) -> bool:
     """Set up config entry."""
+
+    session = aiohttp_client.async_get_clientsession(hass)
+    printer = hass.data[DOMAIN][entry.data[CONF_URL]] = SyncThru(
```

Comment:
Why not use entry ID?
```suggestion
    printer = hass.data[DOMAIN][entry.entry_id] = SyncThru(
```

Reply:
Done, a couple of other places needed the change as well.